op-service/sources/batching/tx_call.go

```diff
@@ -0,0 +1,67 @@
+package batching
+
+import (
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching/rpcblock"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type TxGetByHashCall struct {
+	Abi    *abi.ABI
+	TxHash common.Hash
+	Method string
+}
+
+func NewTxGetByHash(abi *abi.ABI, txhash common.Hash, method string) *TxGetByHashCall {
+	return &TxGetByHashCall{
+		Abi:    abi,
+		TxHash: txhash,
+		Method: method,
+	}
+}
+
+func (b *TxGetByHashCall) ToBatchElemCreator() (BatchElementCreator, error) {
+	return func(block rpcblock.Block) (any, rpc.BatchElem) {
+		out := new(hexutil.Bytes)
+		return out, rpc.BatchElem{
+			Method: "eth_getTransactionByHash",
+			Args:   []interface{}{b.TxHash, block.ArgValue()},
+			Result: &out,
+		}
+	}, nil
+}
+
+func (c *TxGetByHashCall) HandleResult(result interface{}) (*CallResult, error) {
```

Comment:
Should do `DecodeTxParams` inside `HandleResult` ?

Reply:
The `HandleResult` interface requires a return type of `(*CallResult, error)`, so `DecodeTxParams` is implemented separately.

Reply:
`CallResult.out` is of type `[]interface{}`, so should be fine ?

Reply:
Right. The expected result is actually `types.Transaction`, so `DecodeToTx` is moved into `HandleResult` and a `GetTx` method was added to  `CallResult`. `DecodeTxParams` was moved to `UnpackCallData`.

Done in https://github.com/ethstorage/optimism/pull/63/commits/0fbfd260ebfd7cbb03f110ccf8a7b9a5c2f8425b